pkg/odo/cli/component/create.go

```diff
@@ -422,6 +474,15 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 				// Component type: We provide devfile component list to let user choose
 				componentType = ui.SelectDevfileComponentType(catalogDevfileList.Items)
 
+				if co.forceS2i {
```

Comment:
if i am doing `odo create --s2i` in interactive mode, it is still giving me list of devfile components.

Reply:
https://github.com/openshift/odo/pull/3545#discussion_r459221704

Reply:
it is giving only devfile component list, not s2i component. IMO this flag should be disabled for interactive mode.